Algorithm.Framework/Portfolio/AccumulativeInsightPortfolioConstructionModel.cs

```diff
@@ -0,0 +1,243 @@
+                {
+                   // We received a Flat
+
+                   // if adding or subtracting will push past 0, then make it 0
+                   if (Math.Abs(positionSizes[insight.Symbol]) < _percent)
+                   {
+                      positionSizes[insight.Symbol] = 0;
+                   }
+
+                   // otherwise, we flatten by percent
+                   if (positionSizes[insight.Symbol] > 0)
+                   {
+                        positionSizes[insight.Symbol] -= _percent;
+                   }
+                   if (positionSizes[insight.Symbol] < 0)
+                   {
+                      positionSizes[insight.Symbol] += _percent;
+                   }
+                }
+            
+                result[insight] = positionSizes[insight.Symbol];
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Will determine the target percent for each expired insight (since expired insights
+        /// will remove the amount accumulated in portfolio)
+        /// </summary>
+        /// <param name ="activeInsights">The insights that are expiring and must be updated
+        /// <returns>A target percent for each insight</returns>
+        public Dictionary<Insight, double> UpdateExpiredInsights(ICollection<Insight> activeInsights)
+        {
+            var result = new Dictionary<Insight, double>();
+        
+            foreach (var insight in activeInsights)
+            {
+                if (expiredList.ContainsKey(insight))
+                {
+                    continue;
+                }
+                expiredList[insight] = 1;
+
+                // if an expiring insight pushes it past 0, then flatten to 0
+                if ( (Math.Abs(positionSizes[insight.Symbol]) < _percent) && (insight.Direction != 0) )
+                {
+                    positionSizes[insight.Symbol] = 0;
+                }
+                else
+                {
+                    positionSizes[insight.Symbol] -= _percent * (int)insight.Direction;
+                }
+                result[insight] = positionSizes[insight.Symbol];
+            }
+        
+            return result;
+        }
+                
+        /// <summary>
+        /// Create portfolio targets from the specified insights
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="insights">The insights to create portfolio targets from</param>
+        /// <returns>An enumerable of portfolio targets to be sent to the execution model</returns>
+        public override IEnumerable<IPortfolioTarget> CreateTargets(QCAlgorithm algorithm, Insight[] insights)
+        {
+            var targets = new List<IPortfolioTarget>();
+
+            if (algorithm.UtcTime <= _nextExpiryTime &&
+                insights.Length == 0 &&
+                _removedSymbols == null)
+            {
+                return targets;
```

Comment:
The `yield break` statement is equivalent to a `return` statement, except it doesn't return anything, instead it signals that the enumeration of the collection has ended.

conversion:
``` csharp
yield break;
```